### PR TITLE
feat: Jellyfin/Emby server type setup

### PR DESCRIPTION
**Pull request by fallenbagel**
#### Description
This pull request introduces a new server setup for Jellyfin/Emby users. It now mandates the selection of a serverType during the setup process for jellyseerr. Consequently, the previous reliance on an environment variable to define the server type has been deprecated due to its unreliability and hacky nature. This update offers a more dependable and definitive approach to specifying the server type for jellyseerr setup.

Furthermore, there are breaking changes to the API. The media server setup for Jellyfin/Emby users now demands the selection of a serverType (Jellyfin or Emby) during setup, specifically at the endpoint `/api/v1/auth/jellyfin`

#### Screenshot (if UI-related)
Setup page during initial load:
image
Setup Page after selecting Jellyfin:
image
Setup Page after selecting Emby:
image

#### To-Dos

- [x] Successful build `yarn build`
- [ ] Translation keys `yarn i18n:extract`
- [ ] Database migration (if required)

#### Issues Fixed or Closed

- Fixes #XXXX

### CODE REVIEW

**Review by Jumail — DISMISSED** [review]
Everything looks good!

**Comment by sgtcoder** [comment]
Thank you for this. How can I help you on this and test?

**Comment by Fallenbagel** [comment]
> Thank you for this. How can I help you on this and test?

First we need to merge #773. Then this pr will require a refactor to support the new setting schema. Once that's done I'll release a preview tag to test this c:

But for now, you could help us test #773 ;)

(will create a preview tag tonight. Though make sure not to run it on prod, but we want test instances to be done with existing config so duplicate config and test as we want to see if our migration works)

**Comment by sgtcoder** [comment]
I am using Docker, so what I could do is build the image with your preview tag, spin up a new docker container with a copy of my production config folder, and test that way.

**Comment by Fallenbagel** [comment]
> I am using Docker, so what I could do is build the image with your preview tag, spin up a new docker container with a copy of my production config folder, and test that way.

The preview tag will already be built. You'll just have to pull it and try :)

**Comment by Fallenbagel** [comment]
> Sounds good, even better. Let me know when that's ready. I assume it is, but do you have the docker tag?

https://github.com/Fallenbagel/jellyseerr/pull/773#issuecomment-2135013667

**Comment by JaidenW** [comment]
This sounds great and i'm looking forward to this. it would be enough to get me to switch off of ombi, but why not just go all the way and have support for Emby/Jellyfin/Plex all at once. That's my current setup & that would bring this in line with Ombi's level of support

**Comment by Fallenbagel** [comment]
> This sounds great and i'm looking forward to this. it would be enough to get me to switch off of ombi, but why not just go all the way and have support for Emby/Jellyfin/Plex all at once. That's my current setup & that would bring this in line with Ombi's level of support

#100

**Review by Jumail — DISMISSED** [review]
LGTM! 🥳

**Comment by sgtcoder** [comment]
Let me know when ready to test.

**Comment by gauthier-th** [comment]
> Let me know when ready to test pr image when available. Looks like it's `fallenbagel/jellyseerr:preview-jellyfin-emby-server-type`

I just published another preview, you can test it with the `preview-jellyfin-emby-server-type` tag

File: server/routes/auth.ts
Comment (michaelhthomas) [inline server/routes/auth.ts:1]: Should this block on line 418 still be a TODO comment?

```ts
// TODO: If JELLYFIN_TYPE is set to 'emby' then set mediaServerType to EMBY
// if (process.env.JELLYFIN_TYPE === 'emby') {
//   settings.main.mediaServerType = MediaServerType.EMBY;
//   settings.save();
// }
```

Comment (fallenbagel) [reply server/routes/auth.ts:1]: Oh yeah thanks for pointing it out. That should be removed as we are now handling that using the settingsMigrator

**Pull request merged**